Helius sdk

shivaji43 opened the pull request:
# Pull Request Description

## Related Issue
Fixes #157 

## Changes Made
This PR adds the following changes:
<!-- List the key changes made in this PR -->
- Added tool to parse Transactions in human readable format
- Get all assets owned by a Public key
- Create , Get and Delete Webhooks for real time notifications
- Send a transaction with priority fee using Helius API
  
## Implementation Details
<!-- Provide technical details about the implementation -->
- Used Helius API's to implement the changes


## Transaction executed by agent 
<!-- If applicable, provide example usage, transactions, or screenshots -->
![Screenshot 2025-01-07 160320](https://github.com/user-attachments/assets/66327de0-c069-45d4-88d4-3891bef61eca)
![Screenshot 2025-01-07 170044](https://github.com/user-attachments/assets/f90ae856-f43b-4ab6-8905-1b7ad28ac2d7)
![Screenshot 2025-01-09 012651](https://github.com/user-attachments/assets/242db155-c3b8-4bcf-953f-a7a0fe9aada7)
![Screenshot 2025-01-09 012730](https://github.com/user-attachments/assets/d98c1589-5716-4068-bee8-be01cb27b4cc)
![Screenshot 2025-01-09 021915](https://github.com/user-attachments/assets/98ffdf1b-86fb-4ff1-b62f-9eb32d1d0ecb)
![Screenshot 2025-01-10 151839](https://github.com/user-attachments/assets/bb36ca67-d6b3-4cc0-b514-282cc765c6c1)
Transaction id for priority fee 
```34iEaLyeQc71rrv5bwV49LkfxbsYqFeWS3rV7wxKM2MHiNrfj9QxkdfFjkquBdneubyp6PsNVATRSit4ATseGKtH```




## Prompt Used
<!-- If relevant, include the prompt or configuration used -->
```parse this transaction for me 5Mp5eVCdjY2i4uk7oEs3kKSqNab49cauuJANuQhiNq9tWKPRe7QmS65amhTGzYfMGfoNB64kY3Jjzrh8Fy4D8FV6```
```give me assets owned by this address BVdNLvyG2DNiWAXBE9qAmc4MTQXymd5Bzfo9xrQSUzVP limit 3```
```send 0.0001 sol to Eo2ciguhMLmcTWXELuEQPdu7DWZt67LHXb2rdHZUbot7 with high  priority```


## Additional Notes
<!-- Any additional information that reviewers should know -->

## Checklist
- [x] I have tested these changes locally
- [ ] I have updated the documentation
- [x] I have added a transaction link
- [x] I have added the prompt used to test it

shivaji43 commented:
Will implement the remaining in sometime

thearyanag commented on src/tools/get_assets_by_owner.ts:
```diff
@@ -0,0 +1,59 @@
+import { SolanaAgentKit } from "../index";
+import { PublicKey } from "@solana/web3.js";
+import * as dotenv from "dotenv";
+dotenv.config();
```
can you pass Helius API Key from config object instead of taking from env?

thearyanag commented on src/tools/helius_transaction_parsing.ts:
```diff
@@ -0,0 +1,46 @@
+import { SolanaAgentKit } from "../index";
+import * as dotenv from "dotenv";
+dotenv.config();
```
same here, can you pass Helius API Key from config object instead of taking from env?

shivaji43 replied:
Yes i will do that

shivaji43 replied:
yup now i got it

shivaji43 commented:
@thearyanag  I added tool for sending transaction fee with priority fee but it still uses the Transfer tool what should I do?

shivaji43 commented:
@thearyanag  for real time notifications i think webhooks can be more useful , after the user tells  the agent which accounts they want to keep check of , the agent will return a webhook id and webhook url which then the user can use as they want. also the user can edit and delete webhook. 
 what do you think ?

shivaji43 commented:
@thearyanag I have added screenshots for webhook creation, fetching by id and deleting webhooks please take a look at  it

shivaji43 commented:
@thearyanag I've Implemented all the features in the issue please take a look , I have added all the screenshots, if further changes are required I will do them soon

arihantbansal commented on src/tools/send_transaction_with_priority.ts:
This is not needed as a tool. It belongs in utils, and already exists in a form here - https://github.com/sendaifun/solana-agent-kit/blob/main/src/utils/send_tx.ts

I'd suggest you add option to use helius for priority fee calculation if the helius api key is provided in config.

shivaji43 replied:
Ok understood I will do that

shivaji43 replied:
What if I keep it as a tool too as well as I'll change the send_tx.ts to calculate fee as per the helius key?

arihantbansal commented:
Also, add relevant "Action" for each of the added tool - as here https://github.com/sendaifun/solana-agent-kit/tree/main/src/actions

shivaji43 commented:
@arihantbansal  please take a look
